### PR TITLE
feat: implement click callback

### PR DESCRIPTION
**Pull request by powercasgamer**
also updates adventure to 4.13.1

javaw_rFhgS2cem2
test.gif

### CODE REVIEW

File: proxy/velocity.toml
Comment (kashike) [inline proxy/velocity.toml:1]: Remove this file from your PR.

**Review by kashike — CHANGES_REQUESTED** [review]

File: proxy/src/main/java/com/velocitypowered/proxy/command/builtin/VelocityCommand.java [hunk 1]
```diff
@@ -485,4 +488,33 @@ public boolean hasPermission(CommandSource source, String @NonNull [] args) {
     }
 
   }
+
+  /**
+   * Callback SubCommand.
+   */
+  public static class Callback implements SubCommand {
+    private static final Logger logger = LogManager.getLogger(Callback.class);
```
Comment (4drian3d) [inline proxy/src/main/java/com/velocitypowered/proxy/command/builtin/VelocityCommand.java]: This Logger creation is unnecessary, it is not used anywhere.

**Review by 4drian3d — APPROVED** [review]

File: proxy/src/main/java/com/velocitypowered/proxy/provider/ClickCallbackProviderImpl.java [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    }
+
+    public boolean valid() {
+      return hasRemainingUses() && !expired();
+    }
+
+    public UUID id() {
+      return this.id;
+    }
+  }
```
Comment (astei) [inline proxy/src/main/java/com/velocitypowered/proxy/provider/ClickCallbackProviderImpl.java:138]: I'm really not a big fan of this approach.

* We already have an expiring cache, Caffeine. Why have an eviction task running every 5 milliseconds?
* The `HashMap` used here is not thread safe. Velocity is a multi-threaded environment. This kind of implementation would make sense with Paper, with a single server thread, but it won't fly here.

For this reason I am not willing to sign off on this implementation unless these issues were resolved.

**Review by astei — CHANGES_REQUESTED** [review]

**Comment by powercasgamer** [comment]
closing in favor of https://github.com/PaperMC/Velocity/tree/feat/click-callback-provider ?

**Pull request closed without merging**